app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,13 @@
+class WelcomeController < ApplicationController
+  before_action :check_user_logged_in
+  skip_before_action :authenticate_user!, only: [:index]
+
+  def index
+  end
+
+  private
+
+  def check_user_logged_in
+    redirect_to new_user_session_path if user_signed_in?
```

Comment:
Why do we need to redirect user to login page if he is already logged in?

Reply:
A logged in user doesn't have to be on the welcome page as that page is only used to go to sign in page.
So if user is already signed in, I redirected it to login page from where the user admin or simple user will be directed to their respective page defined in after_sign_in_path in application_controller.rb

Reply:
In this case, we'll be sending a request to sign_in action and then it'll send another request to redirect user. Can we do that here according to their roles?